cpp/src/arrow/util/compression.cc

```diff
@@ -155,6 +159,16 @@ Result<std::unique_ptr<Codec>> Codec::Create(Compression::type codec_type,
       break;
 #else
       return Status::NotImplemented("LZ4 codec support not built");
+#endif
+    case Compression::LZ4_FRAME:
+#ifdef ARROW_WITH_LZ4
+      if (compression_level_set) {
+        return Status::Invalid("LZ4 doesn't support setting a compression level.");
```

Comment:
Note: it seems that the LZ4 frame format does support setting compression levels (which internally switches between lz4.h and lz4hc.h). Fine to leave this as is but we might return in the future and enable setting the lz4hc compression levels (values above 0).

Reply:
Ah, right.